zerver/views/users.py

```diff
@@ -247,6 +249,15 @@ def avatar(
         # interact with fake email addresses anyway.
         if is_email:
             raise MissingAuthenticationError()
+
+        if settings.RATE_LIMITING:
+            try:
+                unique_avatar_key = f"{realm.id}/{email_or_id}/{medium}"
```

Comment:
I don't see a way to exploit this, but I feel uneasy about us forming keys here based on completely arbitrary user input here (`email_or_id`). Should we add validation that if it's not an id, it's at least actually an email? And perhaps use a character that's not permitted in the email string as the separator? (`validate_email` seems to allow `/`, but not `:` for example)

@timabbott Thoughts on whether this matters?

Reply:
Your concern in valid, but we already ensure this is not an email just above.


```
        if is_email:
            raise MissingAuthenticationError()
```